aeon/schema/social.py

```diff
@@ -0,0 +1,93 @@
+"""Readers for data relevant to Social experiments."""
+
+from pathlib import Path
+from typing import List, Union
+import json
+
+import pandas as pd
+
+from aeon import util
+import aeon.io.reader as _reader
+
+
+class Pose(_reader.Harp):
+    """Reader for Harp-binarized tracking data given a model that outputs id, parts, and likelihoods."""
+
+    pattern: str
+    columns: List[str] = []
+    extension: str = "bin"
```

Comment:
Do we need to redeclare class fields here instead of calling the base class constructor?

Also, initializing columns to an empty list using this syntax would make the default list shared and mutable across all class instances. I recently came across this issue in the context of data classes: https://peps.python.org/pep-0557/#mutable-default-values

In regular classes this is allowed, but probably not recommended. I realize it wouldn't be a problem the way you are using it now, but might lead to subtle and hard to track bugs down the line.

Calling the base class constructor might also be preferable here so we can use the constructor function scope to guarantee a unique empty list is assigned on creation.

In general I feel this style of declaring and initializing fields is best for data classes where you can use default factories that will be used in the auto-generated constructor to initialize a true default.

I'm happy to consider redefining the architecture of readers using data classes if needed, but in that case I would rather do it for all classes rather than opening exceptions, for consistency of reading (and opening a new issue / discussion).

Reply:
> Also, initializing columns to an empty list using this syntax would make the default list shared and mutable across all class instances.

Thanks for pointing this out. Agree with all this and made this change.